README.md

```diff
@@ -12,30 +12,93 @@ The LightStep distributed tracing library for Go.
 $ go get 'github.com/lightstep/lightstep-tracer-go'
 ```
 
-## Getting started
```

Comment:
nit: It would be nice to split README change into a separate PR. But I don't have strong feelings about that.

Reply:
I'm happy to split all of this out if it's hard to review in one go.